sematic/resolvers/cloud_resolver.py

```diff
@@ -105,6 +132,7 @@ def _detach_resolution(self, future: AbstractFuture) -> str:
 
     def _schedule_future(self, future: AbstractFuture) -> None:
         run = api_client.schedule_run(future.id)
+        self._inc_remote_runs_count()
```

Comment:
If we're *really* going for the server as the source of truth on this count, we shouldn't do this "dead reckoning" of incrementing and decrementing an in-memory counter. Instead, in `_can_schedule_future` we should query the server for all runs which have the same root id as this resolver is handling, and count the number of returned runs which are in the `SCHEDULED` state. If we're just relying on some in-memory state anyway, we might as well just count the number of in-memory futures in the `SCHEDULED` state.

Reply:
Removed the in-memory counter and replaced it with an on-the-fly counting on the number of in-memory futures in the SCHEDULED state.

In order to make this information relevant, a server call should be made to refresh this information. This call is already made in `_wait_for_any_remote_job`, but the information is discarded for all except the first future whose state is detected as changed.

So instead of adding another server call in addition to this partially-ignored one right after it, I made the original call not ignore the rest of the futures. This also has the added benefit of being an optimization that processes all refreshed futures instead of only one in each main SMR loop.